src/ResourceManagement/ContainerInstance/ContainerGroupImpl.cs

```diff
@@ -393,5 +422,29 @@ public string GetLogContent(string containerName, int tailLineCount)
             return (log != null) ? log.Content : "";
         }
 
+        ///GENMHASH:8642FC9B9C8DC7375DEE3CC5736F8853:D8E9D98F2E55B91C74B8A9CCC0D57DFB
+        public ContainerGroupImpl WithDnsPrefix(string dnsPrefix)
+        {
+            if (this.Inner.IpAddress == null)
+            {
+                this.Inner.IpAddress = new IpAddress();
+            }
+            this.Inner.IpAddress.DnsNameLabel = dnsPrefix;
+
+            return this;
+        }
+
+        ///GENMHASH:59348A25FD515049ECD26A6290F76B85:E61328CEA9774DAE97931A8E3F5BA1FA
+        public string DnsPrefix()
+        {
+            return (this.Inner.IpAddress != null) ? this.Inner.IpAddress.DnsNameLabel : null;
```

Comment:
>return (this.Inner.IpAddress != null) ? this.Inner.IpAddress.DnsNameLabel : null; [](start = 12, length = 81)

FYI, this can be done with a single statement:
` return this.Inner.IpAddress?.DnsNameLabel;